src/main/java/com/rtbhouse/kafka/workers/impl/KafkaWorkersImpl.java

```diff
@@ -70,7 +74,22 @@ public KafkaWorkersImpl(
         this.taskFactory = taskFactory;
         this.subpartitionSupplier = new SubpartitionSupplier<>(partitioner);
         this.callback = callback;
-        this.offsetsState = new OffsetsState();
+        this.offsetsState = createOffsetsState(this.config, this.metrics);
+        logger.info("OffsetsState implementation: {}", this.offsetsState.getClass().getName());
+    }
+
+    private static OffsetsState createOffsetsState(WorkersConfig config, WorkersMetrics metrics) {
```

Comment:
should not be necessary to create various types of offset state in case of only one left

Reply:
Done